pkg/apiserver/apiserver.go

```diff
@@ -0,0 +1,329 @@
+
+	// Spec is the spec of the GreptimeDBCluster.
+	Spec *greptimev1alpha1.GreptimeDBClusterSpec `json:"spec,omitempty"`
+
+	// Status is the status of the GreptimeDBCluster.
+	Status *greptimev1alpha1.GreptimeDBClusterStatus `json:"status,omitempty"`
+
+	// Topology is the deployment topology of the GreptimeDBCluster.
+	Topology *GreptimeDBClusterTopology `json:"topology,omitempty"`
+}
+
+// GreptimeDBClusterTopology represents the deployment topology of a GreptimeDBCluster.
+type GreptimeDBClusterTopology struct {
+	// Meta represents the meta component of the GreptimeDBCluster.
+	Meta []*Pod `json:"meta,omitempty"`
+
+	// Datanode represents the datanode component of the GreptimeDBCluster.
+	Datanode []*Pod `json:"datanode,omitempty"`
+
+	// Frontend represents the frontend component of the GreptimeDBCluster.
+	Frontend []*Pod `json:"frontend,omitempty"`
+
+	// Flownode represents the flownode component of the GreptimeDBCluster.
+	Flownode []*Pod `json:"flownode,omitempty"`
+}
+
+// Pod is a simplified representation of a Kubernetes Pod.
+type Pod struct {
+	// Name is the name of the Pod.
+	Name string `json:"name"`
+
+	// Namespace is the namespace of the Pod.
+	Namespace string `json:"namespace"`
+
+	// IP is the IP address of the Pod.
+	IP string `json:"ip"`
+
+	// Node is the name of the node where the Pod is running.
+	Node string `json:"node"`
+
+	// Resource is the resource requirements of the Pod.
+	Resource corev1.ResourceRequirements `json:"resource,omitempty"`
+
+	// Status is the status of the Pod.
+	Status string `json:"status"`
+
+	// StartTime is the time when the Pod started.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+}
+
+// Response represents a response returned by the API.
+type Response struct {
+	// Success indicates whether the request was successful.
+	Success bool `json:"success"`
+
+	// Code is the status code that defined by the Server.
+	Code int `json:"code,omitempty"`
+
+	// Message is additional message returned by the API.
+	Message string `json:"message,omitempty"`
+
+	// Data is the data returned by the API.
+	Data interface{} `json:"data,omitempty"`
+}
+
+// NewServer creates a new Server with the given client and options.
+func NewServer(client client.Client, opts *Options) *Server {
+	return &Server{
+		Client: client,
+		port:   opts.Port,
+	}
+}
+
+// Run starts the HTTP service and listens on the specified port.
+func (s *Server) Run() error {
+	gin.SetMode(gin.ReleaseMode)
+	router := gin.Default()
+
+	api := router.Group(APIGroup)
+
+	// Get all clusters in a specific namespace. If the namespace is 'all', it will return all clusters in all namespaces.
+	api.GET("/namespaces/:namespace/clusters", s.getClusters)
+
+	// Get a specific cluster in a specific namespace.
+	api.GET("/namespaces/:namespace/clusters/:name", s.getCluster)
+
+	klog.Infof("HTTP service is running on port %d", s.port)
+
+	return router.Run(fmt.Sprintf(":%d", s.port))
+}
+
+func (s *Server) getClusters(c *gin.Context) {
```

Comment:
it be better to use listCluster here ?

Reply:
I don't think so. It's coventional to use `get` in REST API.